templates/long.yaml

```diff
@@ -32,3 +32,11 @@
 
 - name: Test Project - Empty
   type: Project
+
+- name: Test Team
+  type: Team
+  can_public_join: false
+  description: Test team description
+  members:
+  - principal_id: 3426116
+  - email: synapseformation-test-user@sagebase.org
```

Comment:
Since we've added support for inviting members, should we also support a `invite_message` attribute? I would normally want to call it `message` to match the [REST](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/MembershipInvitation.html) and [Python](https://python-docs.synapse.org/build/html/index.html#synapseclient.Synapse.invite_to_team) docs, but that seems generic when the context of invitation isn't explicit. 

Alternatively, we could nest `members` and `message` under an `invitations` attribute to provide that context and be a bit more in line with the Synapse API (like Tom said, invitations are a separate request to the API). We could also organize the `invitations` attribute as an array such that invitations with different messages can be issued. 

```suggestion
  invitations:
  - message: Welcome to the Test Team! 
    members:
    - principal_id: 3426116
    - email: synapseformation-test-user@sagebase.org
```

Admittedly, this is a bit verbose, but this is the long template. We could think of ways of expressing this tersely when we work on the short template since most users will probably not even bother with messages. For example, the short template could offer an `invite_members` convenience attribute that takes an array of strings, inferring whether each element is a principal ID (_i.e._integer), email (_i.e._ contains `@`), or username (_i.e._ the rest). We can discuss this further when the short template is opened, but I wanted to quell any concerns with verbosity here for the long template (if there are any).

Reply:
Thanks @BrunoGrandePhD. Do you ever envision different messages sent to different members?  Since each invitation of each member is done in a different rest call, technically each member could have a different invitation message:

```
invitations:
   - prinicipal_id: 3426116
     message: "Welcome to the Test team!"
   - email: synapseformation-test-user@sagebase.org
     message: "Welcome!"
```

I can't really think of a use case that warrants this as you would probably want every person invited to have the same invitation message, but just wanted to see if we wanted to align more closely to the API.

Reply:
I have a hard time imagining each member being invited with a different message, but it's possible that two groups of members (_e.g._ community members and DCC admins) are invited using two different messages. It's admittedly a stretch, but I feel like we should at least support it if the API allows it. 

My above solution is a compromise between the API design (_i.e._ one-to-one mapping between members and messages) and enforcing one message for all listed members. Since most use cases will be covered by one message, I would like to allow multiple members to be listed for a given message as opposed to forcing the same message to repeat itself. If this is in an array, then if the user does need a different message, they can add a second element with a different message and a different set of members (see example below). 

```
invitations:
- message: Invite Message for Group 1!
  members:
  - principal_id: 3426116
  - email: synapseformation-test-user@sagebase.org
- message: Invite Message for Group 2! 
  members:    
  - principal_id: 3948579
  - principal_id: 9895223
```